sites/website/src/docs/advanced/working-with-custom-elements.md

```diff
@@ -0,0 +1,200 @@
+---
+id: working-with-custom-elements
+title: Working with Custom Elements
+sidebar_label: Working with Custom Elements
+keywords:
+  - shadow
+  - DOM
+  - fouc
+  - flash of unstyled content
+  - web components
+---
+
+A good starting resource for understanding Web Components are the [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/Web_Components).
+
+## FOUC & Hiding undefined elements
+
+Custom Elements that have not been upgraded and don't have styles attached can still be rendered by the browser but they likely do not look how they are supposed to. To avoid a [flash of un-styled content (FOUC)](https://webkit.org/blog/66/the-fouc-problem/), visually hide Custom Elements if they have not been defined:
+
+```css
+:not(:defined) {
+  visibility: hidden;
+}
+```
+
+## Updating attributes on the host element
+
+Sometimes you want to affect the host element itself, based on property state. For example, a progress component might want to write various `aria` attributes to the host, based on the progress state. In order to facilitate scenarios like this, you can use a `template` element as the root of your template, and it will represent the host element. Any attribute or directive you place on the `template` element will be applied to the host itself.
```

Comment:
Is it worth mentioning that you only need the `template` element when you need to effect the host element itself?

Reply:
I was thinking it was implied, but I'll specify.